### ISSUE TITLE
Unimported dotted component tags (`<Foo.Bar>`) silently pass with 0 errors

### ISSUE THREAD

**Issue by febergs** [harshmandan/svelte-check-native#39]
## Describe the bug

When a dotted component tag like `<Text.Body>` is used without the root identifier being imported, `svelte-check-native` emits zero diagnostics. Vanilla `svelte-check` 4.7.1 catches every case. This is a parity gap in the "byte-identical diagnostics" promise.

Affects every namespace-style component tag (`<Foo.Bar>`), not just roots that collide with a DOM global. A fabricated namespace that exists nowhere in the codebase or any dependency also passes silently. Bare tags (`<Foo>`) are caught correctly — the gap is specific to dotted tags.

## Reproduction

```svelte
<script lang="ts">
  // No imports
</script>

<Text.Body>crashes at runtime</Text.Body>
<Dialog.Content>same</Dialog.Content>
<Select.Trigger>same</Select.Trigger>
<DropdownMenu.Item>same</DropdownMenu.Item>
<ZzzFakeNamespace.Member>same — fabricated namespace</ZzzFakeNamespace.Member>
```

### Actual — `svelte-check-native` 0.9.6

```
svelte-check-native found 0 errors and 0 warnings
```

### Expected — vanilla `svelte-check` 4.7.1

```
Error: Property 'Body' does not exist on type '{ new (data?: string | undefined): Text; prototype: Text; }'. (ts)
Error: Cannot find name 'Dialog'. (ts)
Error: Cannot find name 'Select'. Did you mean 'onselect'? (ts)
Error: Cannot find name 'DropdownMenu'. (ts)
Error: Cannot find name 'ZzzFakeNamespace'. (ts)

svelte-check found 5 errors and 0 warnings
```

Bare tags are caught by both tools: `<FooBarBaz>` → `Cannot find name 'FooBarBaz'` ✅

## Likely related — v0.3.7 fix

The v0.3.7 fix ("component-instantiation scaffolding for dotted tag names") dropped an early return that was skipping all dotted tags in the analyze phase. Verified on 0.9.6 that it's working for the imported case:
- `<Text.Body totallyBogusProp={123}>` with `Text` imported → ✅ prop error fires
- `<Text.Body>` with `Text` NOT imported → ❌ silent (this bug)

So the v0.3.7 fix unlocked the emit path for dotted tags, but the missing-root case is still broken. The v0.3.7 CHANGELOG mentions that "the root identifier (`UI`) is voided via the existing template-refs pass so the barrel import doesn't trip TS6133" — that voiding is a plausible lead, but I haven't read the Rust source to confirm. Leaving the root-cause diagnosis to whoever knows the emit/template-refs code.

## System Info

- OS: macOS 15
- `svelte-check-native`: 0.9.6
- `svelte-check` (vanilla): 4.7.1
- `svelte`: 5.36.16
- TypeScript engine: tsgo (TypeScript 7 native)
